vernac/egramrocq.ml

```diff
@@ -268,9 +268,9 @@ type (_, _) entry =
 type _ any_entry = TTAny : ('s, 'r) entry -> 's any_entry
 
 let constr_custom_entry : (string, Constrexpr.constr_expr) entry_command =
-  create_entry_command "constr" { eext_fun = (fun s st -> [s], st); eext_eq = (==) (* FIXME *) }
+  create_entry_command "constr" { eext_fun = (fun s st -> [s], st); eext_eq = String.equal }
 let pattern_custom_entry : (string, Constrexpr.cases_pattern_expr) entry_command =
-  create_entry_command "pattern" { eext_fun = (fun s st -> [s], st); eext_eq = (==) (* FIXME *) }
+  create_entry_command "pattern" { eext_fun = (fun s st -> [s], st); eext_eq = String.equal }
```

Comment:
There is nothing in the custom entry mechanism that checks that the names are unique. We only warn in Metasyntax when loading two entries with the same name, but it doesn't prevent diamond situations. As a result this change is wrong but I don't know how much we can abuse it to break parsing, as this equality is only used for rule factorization. I think we should have something storing the dirpath of the current module or something similar in addition to the string.

Reply:
>We only warn in Metasyntax when loading two entries with the same name

When we warn we don't add a duplicate, so there is only ever the one entry with that name.